iFixit/pulldasher
# Change passport-github to passport-github2 and @octokit/rest to v18

This library seems popular and claims to use a different auth method
that I think will bandaid our current brownout issue.

New package: https://github.com/cfsghost/passport-github
Deprecation notice: (Links to original notice)
https://developer.github.com/changes/2020-02-10-deprecating-auth-through-query-param/

---- 

## Changes made:

1. Removed the usage of the npm package `passport-github` in favor of `passport-github2`
2. Updated the npm package `@octokit/rest` to v18
3. By updating `@octokit/rest` , the function calls in `lib/git-manager.js` had to be updated to reflect the new function calls that call the same endpoints (https://github.com/iFixit/pulldasher/pull/211#issuecomment-833121358). Also had to update the `checkOrgMembershipRequirements` function in `lib/authentication.js`
4. Removed the use of the `rate-limit` function as the plugin `@octokit/plugin-throttling` now allows for the GitHub API object to throttle itself (https://github.com/octokit/plugin-throttling.js#readme)
5. Removed the use of the `paginate` function as `@octokit/rest` v18 now has its own pagination function.

## QA

You can run `update-pulldasher` on cominor with a clone of this repo with this branch checked out, and check out https://pulldasher.cominor.com. Look around pulldasher, let it run throughout the day and make sure the cards reflect the state of their PRs at the end of the day.

Associated issue: #210
